lib/galaxy/webapps/galaxy/api/tool_data.py

```diff
@@ -1,22 +1,144 @@
-import os
+from fastapi import (
+    Depends,
+    Path,
+)
+from fastapi.responses import FileResponse
+from fastapi_utils.cbv import cbv
+from fastapi_utils.inferring_router import InferringRouter as APIRouter
 
 from galaxy import (
     exceptions,
-    web
+    web,
 )
+from galaxy.app import StructuredApp
 from galaxy.managers.context import ProvidesAppContext
+from galaxy.managers.tool_data import ToolDataManager
+from galaxy.tools.data._schema import (
+    ToolDataDetails,
+    ToolDataEntryList,
+    ToolDataField,
+)
 from galaxy.web import (
     expose_api,
-    expose_api_raw
+    expose_api_raw,
 )
 from galaxy.webapps.base.controller import BaseAPIController
+from . import (
+    get_admin_user,
+    get_app,
+)
+
+router = APIRouter(tags=['tool data'])
+
+AdminUserRequired = Depends(get_admin_user)
```

Comment:
This little change makes the code read so much more intuitive! `Depends=get_admin_user` - you assume you need the return value, then see it's in the route decorator - so you don't, and then finally see its purpose. `Depends=AdminUserRequired` simply doesn't raise any questions at all. Nice!

Reply:
Glad you like it :)